pylint/checkers/raw_metrics.py

```diff
@@ -107,6 +109,7 @@ def get_type(tokens, start_index):
         line_type = "empty"
     elif i < len(tokens) and tokens[i][0] == tokenize.NEWLINE:
         i += 1
+    line_type = cast(Literal["code", "docstring", "comment", "empty"], line_type)
     return i, pos[0] - start[0] + 1, line_type
```

Comment:
Does `pyright` do this correctly? I think adding an ignore here might be better: this seems like `mypy` failing to understand difficult work flow. By using a cast we never know when it does figure this out and when the cast is unnecessary.

Reply:
Fixed in latest commit